### PR TITLE
DPL-283 Ont sample sheet generation

### LINKED ISSUE #714: DPL-283 - ONT - Generate sample sheet
**User story**
A csv sample sheet needs to be generated for each Sequencing Run which should contain the correct information for uploading to either a GridIon or Promethion

TODO: Talk to Long Read about structure of sample sheet and how they load it

**Who are the primary contacts for this story**
Steve / James

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] The use of similar method to PacBio for generating sample sheets
- [ ] Method on run to generat sample sheet
- [ ] ...

**References**
This story has a relationship with sanger/traction-service#709

**Additional context**

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #714, #892 

Changes proposed in this pull request:

* Added route for ont runs sample sheet
* Added csv generation for ont sample sheet structure 
* Fixed pacbio run design not reading correct ccs value for v11 smrtlink

### CODE REVIEW

File: app/models/ont/min_know_version.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  class MinKnowVersion < ApplicationRecord
+    validates :name, presence: true, uniqueness: true, format: Version::FORMAT
+
+    # This is the other side of the belongs_to association in ont runs.
+    has_many :runs, class_name: 'Ont::Run', dependent: :destroy, inverse_of: :min_know_version
+
+    scope :active, -> { where(active: true) }
+    scope :ordered_by_default, -> { order(default: :desc) }
+
+    # Returns the default SMRT Link version.
```
Comment (stevieing) [inline app/models/ont/min_know_version.rb:14]: Returns the default MinKNOW version

File: app/pipelines/pipelines.rb [hunk 2]
```diff
@@ -57,8 +57,14 @@ def self.find(pipeline)
     send(pipeline.to_s.downcase)
   end
 
+  # Finds all the config files stored in config/pipelines and merges them into a hash
   def self.load_yaml
-    YAML.load_file('config/pipelines.yml')[Rails.env].symbolize_keys
+    config = {}
+    Dir.children('config/pipelines').each do |pipeline_file|
```
Comment (stevieing) [inline app/pipelines/pipelines.rb:63]: Should the path be a constant?

File: app/controllers/v1/ont/runs_controller.rb [hunk 3]
```diff
@@ -7,6 +7,24 @@ module Ont
...
+      def sample_sheet
+        run = ::Ont::Run.find(params[:run_id])
+        csv = run.generate_sample_sheet
+
+        send_data csv,
+                  type: 'text/csv; charset=utf-8; header=present',
+                  disposition: "attachment; filename=#{run.experiment_name}.csv"
+      end
+
+      # endpoint to retrive sample sheet setting
```
Comment (stevieing) [inline app/controllers/v1/ont/runs_controller.rb:21]: Hopefully we won't need this anymore but best to wait on Alex.

File: lib/tasks/create_min_know_versions.rake [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+# This task is used for creating ont_min_know_versions and
+# It is invoked before data migrations to set the version according to the deprecated string version.
+
+namespace :min_know_versions do
+  desc 'Create ONT Min Know versions'
+  task create: :environment do
+    # Can load this from config in the future if need be
+    versions = {
+      v1: {
```
Comment (stevieing) [inline lib/tasks/create_min_know_versions.rake:11]: we can change this to v22. Thankfully versions follow the standard!

File: spec/models/ont/run_spec.rb [hunk 5]
```diff
@@ -4,6 +4,8 @@
 require './spec/support/read_only'
```
Comment (stevieing) [inline spec/models/ont/run_spec.rb:4]: Is spec/support/read_only still needed?

**Review by stevieing — CHANGES_REQUESTED** [review]
Only a couple of minor things. Splitting the config files is much better.

File: app/controllers/v1/ont/runs_controller.rb (hunk 3 above)
Comment (BenTopping) [reply app/controllers/v1/ont/runs_controller.rb]: Ill remove for the time being. Easy enough to restore if it we do need it

**Review by stevieing — APPROVED** [review]

**Pull request merged**